### PR TITLE
fixes, typos and enabling docstests

### PR DESCRIPTION
**Pull request by carlosp420**
This pull request addresses some typos, removed declared but unused variables and enabled docstests in one file.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/__init__.py [hunk 1]
```diff
@@ -382,24 +382,24 @@
...
-from . import PirIO
-from . import SeqXmlIO
-from . import SffIO
-from . import SnapGeneIO
-from . import SwissIO
-from . import TabIO
-from . import QualityIO  # FastQ and qual files
-from . import UniprotIO
-from . import XdnaIO
+from Bio.SeqIO import AbiIO
```
Comment (carlosp420) [inline Bio/SeqIO/__init__.py:385]: it is necessary to use the full path of imports to enable the doctests at the end of the file.

File: Bio/SeqIO/_index.py [hunk 2]
```diff
@@ -342,7 +342,6 @@ def __iter__(self):
         handle = self._handle
         handle.seek(0)
         marker_re = self._marker_re
-        semi_char = b";"
```
Comment (carlosp420) [inline Bio/SeqIO/_index.py:345]: declared but unused variable

File: Bio/SeqIO/_index.py [hunk 3]
```diff
@@ -369,14 +368,13 @@ def __iter__(self):
...
                 # Looks like the pre 2006 style, take first word only
-                # Or, with two colons, the KIPO patent variantion
+                # Or, with two colons, the KIPO patent variation
                 key = line[3:].strip().split(None, 1)[0]
                 if key.endswith(b";"):
                     key = key[:-1]
             else:
                 raise ValueError("Did not recognise the ID line layout:\n%r" % line)
             while True:
-                end_offset = handle.tell()
```
Comment (carlosp420) [inline Bio/SeqIO/_index.py:379]: declared but unused variable

**Comment by peterjc** [comment]
Thank you - I think you're right the utility of quick testing makes absolute imports OK here.

**Pull request merged**